app/components/subject-viewer.cjsx

```diff
@@ -101,7 +118,7 @@ module.exports = React.createClass
       mainDisplay = @renderFrame @state.frame
     else
       mainDisplay = @props.subject.locations.map (frame, index) =>
-        @renderFrame index, {key: "frame-#{index}"}
+        @renderFrame index, {key: "frame-#{index}", progressListener: @progressUpdater.progressUpdateListener, progressMarker: @progressUpdater.progressMarkerRenderer, registerProgressObject: @progressUpdater.progressMonitoredObject}
```

Comment:
Looks like you really should just be passing `progressUpdater` as an argument here.